core-redirects-cae/src/main/java/com/tallence/core/redirects/cae/filter/RedirectMatchingStrategyImpl.java

```diff
@@ -0,0 +1,191 @@
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.*;
+import java.util.function.Function;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static com.coremedia.cap.common.IdHelper.parseContentId;
+import static java.util.Map.Entry.comparingByKey;
+
+/**
+ * Default Strategy to select a redirect in the {@link SiteRedirects} which matches the given request.
+ *
+ * It checks:
+ * <ul>
+ *  <li>if the {@link HttpServletRequest#getPathInfo()} matches the {@link Redirect#getSource()}</li>
+ *  <li>if the {@link HttpServletRequest#getParameterMap()} is matches by all {@link Redirect#getSourceParameters()}</li>
+ * </ul>
+ *
+ * If you want to implement a custom strategy: Try to override {@link #determinePreAction} or {@link #checkUrlParams}.
+ * If you need more methods to be "protected" feel free to create an issue in the gitHub repo.
+ */
+@Service
+public class RedirectMatchingStrategyImpl implements RedirectMatchingStrategy {
+
+  private static final Logger LOG = LoggerFactory.getLogger(RedirectMatchingStrategyImpl.class);
+
+  private final RedirectService redirectService;
+  private final SiteResolver siteResolver;
+
+  public RedirectMatchingStrategyImpl(RedirectService redirectService, SiteResolver siteResolver) {
+    this.redirectService = redirectService;
+    this.siteResolver = siteResolver;
+  }
+
+  @Override
+  public Result getMatchingRedirect(HttpServletRequest request) {
+
+    // Fetch redirects
+    SiteRedirects redirects = getSiteRedirects(request);
+    return determinePreAction(redirects, request);
+  }
+
+  /**
+   * Fetch redirects for the given site.
+   *
+   * @param request current request
+   * @return a result holder
+   */
+  @NonNull
+  private SiteRedirects getSiteRedirects(HttpServletRequest request) {
+    // Determine site (in order to fetch the redirects for it)
+    Site site = getSiteFromRequest(request);
+
+    // Fetch redirect holder for site
+    return redirectService.getRedirectsForSite(site);
+  }
+
+  /**
+   * Because the {@code SiteFilter} might have run or not (depending on the setup), we can either simply take
+   * the current site from the request or we have to parse it outselves.
+   */
+  @Nullable
+  private Site getSiteFromRequest(HttpServletRequest request) {
+    // Check first, if someone has already made the lookup
+    Site site = SiteHelper.getSiteFromRequest(request);
+
+    if (site == null) {
+      // If site is not in request, it means that the SiteFilter has not run yet. So we fetch the site ourselves.
+      // This code is shamelessly copied from SiteFilter.java in cae-base-lib
+      String pathInfo = request.getPathInfo();
+      try {
+        if (!StringUtils.hasLength(pathInfo) || "/".equals(pathInfo)) {
+          LOG.debug("Could not determine a site without a path info in request {}", request);
```

Comment:
Log und Bedingung passen nicht zusammen. Besser wäre:
```suggestion
          LOG.debug("Could not determine a site without a site name in the path info, request: {}", request);
```